### ISSUE TITLE
Line number is not announced in 64 bit version of Microsoft Word

### ISSUE THREAD

**Issue by msuzu15** [nvaccess/nvda#20248]
### Brief summary

NVDA does not announce line number in 64 bit version of Microsoft Word although does announce in 32 bit version of Microsoft Word.

### Steps to reproduce

1. Press Insert + N to open NVDA menu, then select Preferences -> Settings -> Document Formatting and turn on checkbox for Line Numbers.
2. Open microsoft Word.
3. Press Ctrl+N to create new document.
4. Enter "abc" and press Enter key.
5. Enter "def" and press Enter key.
6. Now, document is as below.

abc
def

7. Press up arrow key to move cursor to 2nd line.
8. Press up arrow key to move cursor to 1st line.

### Actual behavior

- NVDA spoke "def" at step 7.
- NVDA spoke "abc" at step 8.

### Expected behavior

- NVDA spoke "Line 2 def" at step 7.
- NVDA spoke "Line 1 abc" at step 8.

### NVDA type

portable copy

### NVDA version

NVDA 2026.1.1

### Windows version

Windows 11 Enterprise version 25H2

### Name and version of other software in use when reproducing the issue

Word for Microsoft 365 64 bit version

### Other information about your system

This problem does not appear with 32 bit version of Word for Microsoft 365.

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the System Accessibility Repair Tool (COM Registration Fixing Tool in older versions) in NVDA's tools menu?

I have not run the System Accessibility Repair Tool

**Comment by msuzu15** [comment]
Found following workaround for 64 bit of Word: (NVDA still speaks line number in 32 bit version of Word without below step)
- On Word screen, press F6 key several time to move focus to status bar.
- Press Shift+F10 to open context menu, then tick Line Number so that line number is shown on status bar.

**Comment by michaelDCurran** [comment]
Recent versions of MS Word suppress providing line numbers to assistive technologies unless they are visually shown: Layout -> line numbers.
For me, with NVDA 64 bit and MS Word 64 bit, if I turn on line numbers in both NvDA and MS Word, then NvDA reports them.
Closing as works for me. Please test with a 32 bit version of NvDA such as 2025.3 and open if line numbers read there for you, but not with 2026.1.

**Comment by michaelDCurran** [comment]
To clarify, it is not the difference between 32 and 64 bit, but the build version of MS Word. Please provide the full MS Word version strings for  the 32 bit and 64 bit versions you tested.

**Comment by CyrilleB79** [comment]
> Recent versions of MS Word suppress providing line numbers to assistive technologies unless they are visually shown: Layout -> line numbers. For me, with NVDA 64 bit and MS Word 64 bit, if I turn on line numbers in both NvDA and MS Word, then NvDA reports them.

So if I understand correctly, the behaviour is not consistent between apps. Some need to have line number displayed visually to have NVDA report them (e.g. new Word), while other apps can have line number reported by NVDA even without displaying anything (e.g. log reader).

There are also apps where line number does not seem to work, e.g. Windows 11 Notepad.

@michaelDCurran can you confirm I am correct and, if yes, reopen the issue to have this documented?

**Comment by seanbudd** [comment]
@CyrilleB79 - is this something you think needs to be documented in Word, the User Guide or our basic training materials? My read of this, is that it's not appropriate for the User Guide, and it is really up to the software themselves to document how they work in this way. I could see some value in adding to our Word basic training, but the content could possibly get dated fast.

**Comment by msuzu15** [comment]
> To clarify, it is not the difference between 32 and 64 bit, but the build version of MS Word. Please provide the full MS Word version strings for the 32 bit and 64 bit versions you tested.

Actually I tested Microsoft® Word 2019 MSO (バージョン 2508 ビルド 16.0.19127.20302) 32 bit.
I just installed and tested Microsoft® Word for Microsoft 365 MSO (バージョン 2605 ビルド 16.0.20026.20140) 32 bit, then I got same result as Microsoft® Word for Microsoft 365 MSO (バージョン 2604 ビルド 16.0.19929.20172) 64 bit.

**Comment by msuzu15** [comment]
> Recent versions of MS Word suppress providing line numbers to assistive technologies unless they are visually shown: Layout -> line numbers. 

I don't like to use Layout - Line Numbers because it means line number is printed to the document.  I just wish NVDA announces line number so that I can know which line I'm reading.

> For me, with NVDA 64 bit and MS Word 64 bit, if I turn on line numbers in both NvDA and MS Word, then NvDA reports them. 

Below steps worked for both 32 bit and 64 bit of Word to let NVDA announce line number.

- On Word screen, press F6 key several time to move focus to status bar.
- Press Shift+F10 to open context menu, then turn on checkbox for Line Number so that line number is shown on status bar.

**Comment by CyrilleB79** [comment]
> [@CyrilleB79](https://github.com/CyrilleB79) - is this something you think needs to be documented in Word, the User Guide or our basic training materials? My read of this, is that it's not appropriate for the User Guide, and it is really up to the software themselves to document how they work in this way.

I was thinking to the User Guide, in Document formatting settings section. The goal would be to avoid questions / bug reports such as: "I have enabled line number reporting but they are still not reported by NVDA". Such question arises because:
* in some applications / edit fields, line numbers can be reported only if the application shows it visually. That is NVDA provide information that is also accessible to sighted people
* in other app / edit fields, line numbers can be reported by NVDA even if the edit field does not show them for sighted people. That is: NVDA provides an additional feature, not accessible to sighted people, considering that line number can compensate the fact that blind people may have more difficulty than sighted people to figure where they are in a page/document or how to turnb back to a specific position.
